docs/reference/mapping/types/search-as-you-type.asciidoc

```diff
@@ -0,0 +1,267 @@
+
+When adding a field of this type to a mapping
+
+[source,js]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "properties": {
+      "my_field": {
+        "type": "search_as_you_type"
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+
+This creates the following fields
+
+[horizontal]
+
+`my_field`::
+
+    Analyzed as configured in the mapping. If an analyzer is not configured,
+    the default analyzer for the index is used
+
+`my_field._2gram`::
+
+    Wraps the analyzer of `my_field` with a shingle token filter of shingle
+    size 2
+
+`my_field._3gram`::
+
+    Wraps the analyzer of `my_field` with a shingle token filter of shingle
+    size 3
+
+`my_field._index_prefix`::
+
+    Wraps the analyzer of `my_field._3gram` with an edge ngram token filter
+
+
+Setting a larger `max_shingle_size` in the mapping will create shingle fields
+with larger sized shingles, of 2 up to `max_shingle_size`. The
+`my_field._index_prefix` subfield will always use the analyzer from the
+subfield with the largest shingle size. For example the mapping
+
+[source,js]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "properties": {
+      "my_field": {
+        "type": "search_as_you_type",
+        "max_shingle_size": 4
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+
+Will create the fields as such. Note how this changes how the
+`my_field._index_prefix` field is analyzed
+
+[horizontal]
+
+`my_field`::
+
+    Analyzed as configured in the mapping. If an analyzer is not configured,
+    the default analyzer for the index is used
+
+`my_field._2gram`::
+
+    Wraps the analyzer of `my_field` with a shingle token filter of shingle
+    size 2
+
+`my_field._3gram`::
+
+    Wraps the analyzer of `my_field` with a shingle token filter of shingle
+    size 3
+
+`my_field._4gram`::
+
+    Wraps the analyzer of `my_field` with a shingle token filter of shingle
+    size 4
+
+`my_field._index_prefix`::
+
+    Wraps the analyzer of `my_field._4gram` with an edge ngram token filter
+
+
+The same input text is indexed into each of these fields automatically, with
+their differing analysis chains, when an indexed document has a value for the
+root field `my_field`.
+
+[source,js]
+--------------------------------------------------
+PUT my_index/_doc/1?refresh
+{
+  "my_field": "quick red fox lazy brown dog"
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+To search for documents with values of `my_field` that can complete the
+query text, use a `match_phrase_prefix` query
+
+[source,js]
+--------------------------------------------------
+GET my_index/_search
+{
+  "query": {
+    "match_phrase_prefix": {
+      "my_field": "red f"
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
```

Comment:
Since we've decided to not expand the search automatically to the shingle field I think we need an example with a `multi_match` that targets the root field and the shingle field with the type set to `match_phrase_prefix`, WDYT ?